From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ian Wahbe <ian@wahbe.com>
Date: Tue, 4 Apr 2023 09:19:57 -0700
Subject: [PATCH] fork


diff --git a/pkg/providers/property/resource_akamai_property.go b/pkg/providers/property/resource_akamai_property.go
index 9c66c004..a32b229d 100644
--- a/pkg/providers/property/resource_akamai_property.go
+++ b/pkg/providers/property/resource_akamai_property.go
@@ -517,9 +517,10 @@ func resourcePropertyCreate(ctx context.Context, d *schema.ResourceData, m inter
 	// Save minimum state BEFORE moving on
 	d.SetId(propertyID)
 	attrs := map[string]interface{}{
-		"group_id":    groupID,
-		"contract_id": contractID,
-		"product_id":  productID,
+		"group_id":       groupID,
+		"contract_id":    contractID,
+		"product_id":     productID,
+		"latest_version": 1, // we are defaulting to 1 here in the create like we do below in the property creation
 	}
 	if err := tf.SetAttrs(d, attrs); err != nil {
 		return diag.FromErr(err)
@@ -789,6 +790,7 @@ func resourcePropertyUpdate(ctx context.Context, d *schema.ResourceData, m inter
 
 	var propertyVersion int
 	if v, ok := d.GetOk("read_version"); ok && v.(int) != 0 {
+		property.LatestVersion = v.(int)
 		propertyVersion = v.(int)
 	} else {
 		propertyVersion = property.LatestVersion
